code/game/objects/structures/vehicles/lightcycle.dm

```diff
@@ -0,0 +1,244 @@
+	overlays += light_overlay
+
+/obj/structure/bed/chair/vehicle/lightcycle/can_apply_inertia()
+	return FALSE
+
+/obj/structure/bed/chair/vehicle/lightcycle/Process_Spacemove(var/check_drift = 0)
+	return TRUE
+
+/obj/structure/bed/chair/vehicle/lightcycle/set_keys()
+	return
+
+/obj/structure/bed/chair/vehicle/lightcycle/lock_atom(var/atom/movable/AM)
+	..()
+	update_icon()
+	trigger_movement()
+
+/obj/structure/bed/chair/vehicle/lightcycle/unlock_atom(var/atom/movable/AM)
+	to_chat(occupant, "<span class='notice'>You begin dismounting \the [src]...")
+	spawn(5)	//to prevent riders from just getting off the cycle to avoid hitting obstacles
+		if(!gcDestroyed)
+			..()
+			update_icon()
+			dismount(AM)
+
+/obj/structure/bed/chair/vehicle/lightcycle/proc/dismount(mob/user)
+	to_chat(user, "<span class='notice'>As you dismount \the [src], it dissolves into nothing.</span>")
+	if(summoning_rod)
+		summoning_rod.icon_state = initial(summoning_rod.icon_state)
+		summoning_rod.summoned_cycle = null
+		summoning_rod.cycle_active = FALSE
+		summoning_rod.update_icon()
+		summoning_rod.paired_to = null
+		summoning_rod = null
+	qdel(src)
+
+/obj/structure/bed/chair/vehicle/lightcycle/proc/trigger_movement()
+	while(occupant)
+		movement_process()
+		sleep(1)
+
+/obj/structure/bed/chair/vehicle/lightcycle/proc/movement_process()
+	if(!occupant)
+		return
+	if(occupant.incapacitated() || destroyed)
+		unlock_atom(occupant)
+		return
+	if(!check_key(occupant))
+		unlock_atom(occupant)
+		return
+	var/turf/T = get_turf(src)
+	if(!T.has_gravity())
+		if(!Process_Spacemove(0))
+			return
+
+	var/can_pull_tether = 0
+	if(occupant.tether)
+		if(occupant.tether.attempt_to_follow(occupant,get_step(src,dir)))
+			can_pull_tether = 1
+		else
+			var/datum/chain/tether_datum = occupant.tether.chain_datum
+			tether_datum.snap = 1
+			tether_datum.Delete_Chain()
+
+	step(src, dir)
+
+	if(occupant)
+		if(T != loc)
+			var/mob/living/L = occupant
+			if(istype(L))
+				L.handle_hookchain(dir)
+
+		if(occupant.tether && can_pull_tether)
+			occupant.tether.follow(occupant,T)
+			var/datum/chain/tether_datum = occupant.tether.chain_datum
+			if(!tether_datum.Check_Integrity())
+				tether_datum.snap = 1
+				tether_datum.Delete_Chain()
+
+	spawn(1)
+		if(delay_ribbon)
+			delay_ribbon--
+		else
+			new /obj/lightribbon(T,l_color,lastdir,lastLASTdir)
+			lastLASTdir = lastdir
+			lastdir = dir
+
+	update_mob()
+
+/obj/structure/bed/chair/vehicle/lightcycle/relaymove(var/mob/living/user, direction)
+	if(!check_key(user))
+		to_chat(user, "<span class='notice'>You'll need the keys in one of your hands to drive \the [src].</span>")
+
+	if(direction != turn(dir, 180))
+		dir = direction
+		user.dir = direction
+
+	update_mob()
+
+/obj/structure/bed/chair/vehicle/lightcycle/Bump(atom/A)
+	if(istype(A, /mob))
+		var/mob/M = A
+		src.visible_message("<span class='warning'>\The [occupant] knocks over \the [M] with \his [src.name]!</span>")
+		M.stop_pulling()
```

Comment:
I took it from the code executed when a MULE runs over a mob, I don't know how old it is.